cassandra/cluster.py

```diff
@@ -2252,8 +2270,9 @@ def prepare(self, query, custom_payload=None):
             log.exception("Error preparing query:")
             raise
 
+        prepared_keyspace = message.keyspace if message.keyspace else self.keyspace
```

Comment:
This will always be `self.keyspace` since the message is built like `message = PrepareMessage(query=query)`. We can add the `keyspace` parameter to `session.prepare` but that's probably a different ticket.